### PR TITLE
handle unknown subsections in the name section

### PR DESCRIPTION
**Pull request by mangas**
Closes https://github.com/paritytech/parity-wasm/issues/271

### CODE REVIEW

**Comment by athei** [comment]
Please satisfy the CI. The PR has a lot of noise by pointless stylistic changes.

**Comment by mangas** [comment]
@athei I think I fixed the remaining ones, can you approve the CI run again please?

**Comment by athei** [comment]
This is basically about new subsections in the name section that are not supported are passed through verbatim without parsing?

**Comment by athei** [comment]
> @athei I think I fixed the remaining ones, can you approve the CI run again please?

Just run `cargo +nightly fmt`.

**Comment by mangas** [comment]
> > @athei I think I fixed the remaining ones, can you approve the CI run again please?
> 
> Just run `cargo +nightly fmt`.

I did, I think my IDE was on the backgroudn reformatting it 😆

**Comment by mangas** [comment]
> This is basically about new subsections in the name section that are not supported are passed through verbatim without parsing?

Correct

**Comment by mangas** [comment]
@athei any other changes needed? 🙏

File: src/elements/name_section.rs [hunk 1]
```diff
@@ -97,7 +97,10 @@ impl NameSection {
 					local_names = Some(LocalNameSubsection::deserialize(module, rdr)?);
 				},
 
-				_ => return Err(Error::UnknownNameSubsectionType(subsection_type)),
+				_ => {
+					let mut buf = vec![0; size];
+					rdr.read(&mut buf)?;
+				},
```
Comment (athei) [inline src/elements/name_section.rs:103]: The unknown subsection is just thrown away, right? Maybe add a comment here explaining that.

Comment (mangas) [reply src/elements/name_section.rs]: comment added

**Comment by mangas** [comment]
@athei do you think we have enough to merge this PR?

**Review by athei — APPROVED** [review]

**Comment by athei** [comment]
> @athei do you think we have enough to merge this PR?

Yes from my point of view. Need another review, though.

**Comment by mangas** [comment]
Is that from @pepyakin? Anyone else we can ping here?

**Comment by pepyakin** [comment]
Interesting. According to the [wasm spec](https://webassembly.github.io/spec/core/appendix/custom.html), this is indeed an incorrect names section. 

Apparently, those new items are coming from the [extended-name-section](https://github.com/WebAssembly/extended-name-section) proposal. 

I guess we can ignore those for now and proceed with merging this PR.

**Comment by athei** [comment]
You still need to approve / merge. I can't.

**Review by pepyakin — APPROVED** [review]

File: src/elements/name_section.rs [hunk 2]
```diff
@@ -97,7 +97,12 @@ impl NameSection {
 					local_names = Some(LocalNameSubsection::deserialize(module, rdr)?);
 				},
 
-				_ => return Err(Error::UnknownNameSubsectionType(subsection_type)),
```
Comment (axic) [inline src/elements/name_section.rs:100]: Is this error still emitted anywhere? Perhaps would be a cleanup to drop it from the enum.

**Comment by mangas** [comment]
@athei @pepyakin thanks for the help so far.

Would be possible to create a new tag with the changes so I can use them in https://github.com/paritytech/wasm-instrument/pull/16?

File: src/elements/name_section.rs (hunk 2 above)
Comment (athei) [reply src/elements/name_section.rs:100]: That is a good idea if this is the case.

Comment (mangas) [reply src/elements/name_section.rs:100]: seems like it was only emitted by the code we just removed. 

https://github.com/paritytech/parity-wasm/pull/330

**Pull request merged**